app/models/device_story/device.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module DeviceStory
+  class Device < ApplicationRecord
+    self.table_name = 'device_story_devices'
+
+    validates :device_urn, presence: true, uniqueness: true
+    validates :metadata, presence: true
+
+    before_save :update_from_metadata
+
+    private
+
+    def update_from_metadata
+      self.device_id = metadata['device']
```

Comment:
Scenario to consider:

New device metadata arrives, but the metadata has no points. In this scenario, last_location will remain populated with the old last_location, however, the database row will still be updated.

For last_values, if there is no last_values, `self.last_values` will correctly be set to `nil`. For consistency, I think `self.last_location` should also be set to `nil` in this situation.

Reply:
per today's recording, let's just not update if we don't have a lat/lon

@jamoross makes a good point that we probably shouldn't store 0,0 either since it's likely wrong. Or at very least we should log it or something.